test/Common/DurableTaskEndToEndTests.cs

```diff
@@ -4431,6 +4431,69 @@ public async Task Purge_Single_Instance_History(bool extendedSessions, string st
             }
         }
 
+        [Theory]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        [MemberData(nameof(TestDataGenerator.GetFullFeaturedStorageProviderOptions), MemberType = typeof(TestDataGenerator))]
+        public async Task GetStatusAsyncMultipleInstances(string storageProvider)
+        {
+            const string testName = nameof(this.GetStatusAsyncMultipleInstances);
+            using (ITestHost host = TestHelpers.GetJobHost(this.loggerProvider, testName, false, storageProviderType: storageProvider))
+            {
+                await host.StartAsync();
+
+                var client1 = await host.StartOrchestratorAsync(nameof(TestOrchestrations.Counter), 1, this.output);
+                var client2 = await host.StartOrchestratorAsync(nameof(TestOrchestrations.Counter), 2, this.output);
+
+                string firstInstanceId = client1.InstanceId;
+                string secondInstanceId = client2.InstanceId;
+                string thirdInstanceId = "00000000";
+
+                var instanceIdList = new List<string> { firstInstanceId, secondInstanceId, thirdInstanceId };
+
+                var statusList = await client1.InnerClient.GetStatusAsync(instanceIdList, showHistory: false, showHistoryOutput: false, showInput: true);
+                Assert.Equal("1", statusList[0].Input.ToString());
+                Assert.Equal("2", statusList[1].Input.ToString());
+                Assert.Null(statusList[2]);
+            }
+        }
+
+        [Theory]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        [MemberData(nameof(TestDataGenerator.GetBooleanAndFullFeaturedStorageProviderOptions), MemberType = typeof(TestDataGenerator))]
+        public async Task PurgeMultipleInstanceHistory(bool extendedSessions, string storageProvider)
+        {
+            using (ITestHost host = TestHelpers.GetJobHost(
+               this.loggerProvider,
+               nameof(this.PurgeMultipleInstanceHistory),
+               extendedSessions,
+               storageProviderType: storageProvider))
+            {
+                await host.StartAsync();
+
+                var client1 = await host.StartOrchestratorAsync(nameof(TestOrchestrations.SayHelloWithActivity), "foo", this.output);
+                var client2 = await host.StartOrchestratorAsync(nameof(TestOrchestrations.SayHelloWithActivity), "bar", this.output);
+                var client3 = await host.StartOrchestratorAsync(nameof(TestOrchestrations.SayHelloWithActivity), "baz", this.output);
+
+                string firstInstanceId = client1.InstanceId;
+                string secondInstanceId = client2.InstanceId;
+                string thirdInstanceId = client3.InstanceId;
+                var instanceIdList = new List<string> { firstInstanceId, secondInstanceId, thirdInstanceId };
+
+                await client1.InnerClient.PurgeInstanceHistoryAsync(instanceIdList);

```

Comment:
Should we add a call to `WaitForCompletionAsync` to ensure the orchestrations actually terminate prior to purging their history? I worry that otherwise we risk introducing a flaky test. This applies to both tests.